plugins/UsersManager/Model.php

```diff
@@ -254,6 +254,12 @@ public function getUserByTokenAuth($tokenAuth)
         return $db->fetchRow('SELECT * FROM ' . $this->table . ' WHERE token_auth = ?', $tokenAuth);
     }
 
+    public function getUserByNotificationToken($token)
+    {
+        $db = $this->getDb();
+        return $db->fetchRow('SELECT * FROM ' . $this->table . ' WHERE notification_token = ?', $token);
```

Comment:
Can we store the token hashed ideally? It would be great to have this hashed in case there is a SQL injection, or a DB gets hacked, etc.  see https://www.php.net/manual/en/function.password-hash.php 

we could reuse `Piwik\Auth\Password`

We've always had the trouble with the `token_auth` being in plain text which we will now finally hash in Matomo 4 and be great to avoid storing passwords in plain text in the future. 

It means we need to generate the token when we install the service worker basically. If we don't have the token at a certain time when we need it, then we could possibly regenerate the token and update the service worker.

It would be fine to store the token in the session for a very short time I suppose (only if really needed but ideally not since sessions are stored in the DB).